src/libasr/pass/global_stmts.cpp

```diff
@@ -100,6 +100,11 @@ void pass_wrap_global_stmts_into_function(Allocator &al,
                 throw LCompilersException("Unsupported type of global scope node");
             }
         }
+        std::vector<std::string> symbols;
+        unit.m_global_scope->move_symbols_from_global_scope(fn_scope, symbols);
+        for (auto &a: symbols) {
+            unit.m_global_scope->erase_symbol(a);
+        }
```

Comment:
I would not do this every time. Either as an option (via an argument), or even better, refactor this into a function and let the frontend call it. Add verify() in this refactored function.

Reply:
You can add this function into `global_stmts.h`, so there will be two separate functions: one moves the global statements/expressions, the other moves the global variables.